networks/rpc/client.go

```diff
@@ -638,6 +710,104 @@ func (c *Client) dispatch(conn net.Conn) {
 	}
 }
 
+// dispatchGorillaWS is the main loop of the client that uses Gorilla websocket.
+// It sends readGorillaWS messages to waiting calls to Call and BatchCall
+// and subscription notifications to registered subscriptions.
+func (c *Client) dispatchGorillaWS(conn *gorillaws.Conn) {
```

Comment:
What about combining `dispatch` and `dispatchGorillaWS` functions together? I think quite a few lines are dulicated.